joomla/joomla-cms
# Track the package/extension relationships

### Summary of Changes

Presently there is no tracking in the database of a relationship between a package extension and the extensions it installs, really the only way to know this information is to process the package extension's XML manifest.  This is a step at improving this so we can do other things throughout the API.

### Testing Instructions

With the patch applied, installing a package extension (language packages are great for testing here) should record the package extension's ID to the `#__extensions` table's new `package_id` column.  A query failure setting this info is gracefully handled and essentially results in the current status.

This query will be run during the `finaliseInstall` step of the install process.  This step is run by the `install` and `update` paths (`discover_install` as well but packages don't support this so it's a mute point), so as extension packages are updated after this change is deployed, packages will start to correctly associate themselves with child extensions.

### Documentation Changes Required

Note the newly added database column.

### TODO

- [x] Add install schema for the rest of the drivers
- [x] Add the update schema for all drivers
